controllers/cinder_controller.go

```diff
@@ -637,6 +637,19 @@ func (r *CinderReconciler) generateServiceConfigMaps(
 	templateParameters["ServiceUser"] = instance.Spec.ServiceUser
 	templateParameters["KeystonePublicURL"] = authURL
 
+	// Select CephBackend
+	cephClient := cinder.GetCephBackend(instance)
+
+	/** If CephBackend info are passed, populate the required templateParameters
+	to make sure Cinder is able to interact with an external Ceph cluster
+	using the Client Key provisioned on Ceph
+	**/
+	if cephClient {
+		templateParameters["CephClusterFSID"] = instance.Spec.CephBackend.CephClusterFSID
+		templateParameters["CephClusterMonHosts"] = instance.Spec.CephBackend.CephClusterMonHosts
+		templateParameters["CephClientKey"] = instance.Spec.CephBackend.CephClientKey
```

Comment:
If `cephClient` is `false` and these parameters are then not set in the `templateParameters` map, does the template processing work?  I don't recall if the templating mechanism gracefully ignores `nil` template variable references or throws an exception.

Reply:
Answered my own question: seems to be fine.  Tried the new Glance operator and I see that it just dumps `<no value>` in for missing keys:

```
$ oc rsh glance-6dff4b4dd7-kb887
sh-5.1# cat /var/lib/config-data/merged/ceph.conf 
[global]
fsid = <no value>
mon_host = <no value>
```

Reply:
and since we control `enabled_backends`, that section and those files are totally ignored if no ceph.